fix: update CJS tsconfig moduleResolution to nodenext

braianj opened the pull request:
This PR migrates the package to ESM-only, removing the CommonJS build.

## Breaking Change ⚠️

**This is a breaking change for consumers using CommonJS (`require()`).**

The package now only supports ES Module imports:
// ✅ Supported
import { useAsyncState } from '@dcl/hooks'

// ❌ No longer supported
const { useAsyncState } = require('@dcl/hooks')

## Changes

- Added `"type": "module"` to `package.json`
- Updated `main` and `exports` to ESM-only
- Added explicit `types` field in exports for better TypeScript support
- Removed `tsconfig.cjs.json`
- Simplified build scripts

braianj committed:
fix: update CJS tsconfig moduleResolution to nodenext

LautaroPetaccio commented on tsconfig.cjs.json:
If we don't have issues with old webpack versions, this library, which is intended to be used only in the browser and is new, shouldn't build a CommonJS version. The only thing to take into consideration would be the tests in the different sites, where we would need to pre-process those dependencies before running the tests.

braianj committed:
chore: migrate to ESM-only build

BREAKING CHANGE: CJS build has been removed. This package now only
supports ESM imports.

- Add type: module to package.json
- Remove CJS build configuration and output
- Update exports to ESM-only with proper types declaration
- Simplify build scripts

LautaroPetaccio commented on package.json:
```diff
@@ -1,22 +1,21 @@
+  "type": "module",
+  "main": "esm/index.js",
   "module": "esm/index.js",
   "exports": {
```
The `exports` should not be used anymore.

braianj committed:
feat: remove export from package.json
